File: 643/maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var findMaxAverage = function (nums, k) {
    let maxAverage = -Infinity;
    let n = nums.length;
    for (let start = 0; start <= n - k; start++) {
        let sum = 0;
        for (let i = start; i < start + k; i++) {
            sum += nums[i];
        }
        if (sum / k > maxAverage) {
            maxAverage = sum / k;
        }
    }
    return maxAverage;
};

File: 643/MaximumAverageSubarrayI.scala
object MaximumAverageSubarrayI {
  def findMaxAverage(nums: Array[Int], k: Int): Double = {
    var maxAvg = Double.MinValue
    val len = nums.length
    for (start <- 0 to len - k) {
      var sum: Double = 0
      for (i <- start until start + k) {
        sum += nums(i)
      }
      if (sum/k > maxAvg) maxAvg = sum/k
    }
    maxAvg
  }

  def main(args: Array[String]): Unit = {
    val nums = Array(1,12,-5,-6,50,3)
    val k = 4
    println(findMaxAverage(nums, k))
  }
}
